### PR TITLE
feat: (experimental) partial support for Svelte v5 parser

### PR DESCRIPTION
**Pull request by ota-meshi**
related to sveltejs/eslint-plugin-svelte#587,  #423

~~This PR is still a work in progress. It must support at least the following:~~

- [x] Returns the same AST as before using the new Svelte parser.
- [ ] ~~Support for Runes~~. Listed in #423
  - [ ] ~~Add Runes symbol to global variable~~. Listed in #423
  - [ ] ~~Apply correct type information to $derived~~. Listed in #423
  - [ ] ~~Make TypeScript files parsable~~. Listed in #423

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "target": "es2015",
     "strict": true,
     "lib": ["es2020", "DOM"],
-    "module": "commonjs",
+    "module": "Node16",
```
Comment (baseballyama) [inline tsconfig.json:6]: If we change this, we can not pass the CI with Node14. But can we drop it?
Svelte4 need at least Node16.
(Then we can merge https://github.com/sveltejs/svelte-eslint-parser/pull/399)

Comment (ota-meshi) [reply tsconfig.json:6]: Oh, I did not know that. I'll revert it.

**Review by baseballyama — APPROVED** [review]
I think it's LGTM. Thank you!
But how we will release it?
We will go to same way as `prettier-plugin-svelte`?
(Just release minor version.)

https://github.com/sveltejs/prettier-plugin-svelte/releases/tag/v3.1.0

**Comment by ota-meshi** [comment]
I'd like to release it at v0.34 if it doesn't contain breaking changes, but since it's still v0, minor releases may also require people to explicitly use the new version.
After ESLint v9 is released, I would like to release a major version of this parser and eslint-plguin-svelte.

**Pull request merged**